tests/spectron/runner.spec.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const Bluebird = require('bluebird')
+const spectron = require('spectron')
+const path = require('path')
+
+describe('Spectron', function () {
+  this.timeout(15000)
+
+  beforeEach(function () {
+    this.app = new spectron.Application({
+      path: path.join(__dirname, '..', '..', 'node_modules', '.bin', 'electron'),
```

Comment:
Would there be any way to parameterise / override this path, so that these same Spectron tests could also be used to test the AppImage etc?  (and would this fit in better with the Makefile if it used `npx electron`, or even `npm start`?)

Reply:
I don't think we will ever be able to use Spectron to test an AppImage. The AppImage runtime will run the predefined binary as soon as we execute it, where with Spectron, we would need to run the Electron binary that is embedded inside the AppImage directly, bypassing the runtime.

Reply:
I've just double-checked, and the test spectron script I posted in #1870 definitely works with both the .deb installed version of Etcher and also the AppImage version of Etcher.

Reply:
Oh, wow. I wonder how that works. Let me see how can easily parameterise this.

Reply:
Done!